### PR TITLE
Refresh token expiration #37

### PR DESCRIPTION
**Pull request by atwoosnam**
See recent comments in #37 for some demos that these changes work

### CODE REVIEW

File: src/Services/src/CfCli/CfCliService.cs [hunk 1]
```diff
@@ -98,6 +99,23 @@ public async Task<Version> GetApiVersion()
             }
         }
 
+        /// <summary>
+        /// Retrieves an access token to be used for authentication with Cloud Foundry UAA.
+        /// <para>
+        /// A cached value of the access token will be returned as long as the expiration time of the token has not yet been reached.
+        /// If there is no cached value for the access token or the expiration time of the cached token has been reached, this method 
+        /// will invoke the CF CLI oauth-token command (<see cref="_getOAuthTokenCmd"/>) to acquire a new access token. Invoking
```
Comment (ccheetham) [inline src/Services/src/CfCli/CfCliService.cs:107]: General thoughts on what gets documented.  The "see cref" targets an internal scoped var.  In general, API docs are for API clients who typically cannot access internal/private things.  If, however, these docs are for team members, then doc'ing private/internal stuff is cool.

File: src/Services/src/CfCli/CfCliService.cs [hunk 2]
```diff
@@ -134,7 +154,12 @@ public string GetOAuthToken()
                         }
                         catch (Exception ex)
                         {
-                            _logger.Error($"Something went wrong while attempting to renew access token {ex.ToString()}");
+                            if (ex.GetType() == typeof(InvalidRefreshTokenException))
```
Comment (ccheetham) [inline src/Services/src/CfCli/CfCliService.cs:157]: prefer `if (ex is InvalidRefreshTokenException)`

**Review by ccheetham — CHANGES_REQUESTED** [review]
Many (most? all?) of the new logging statements use interpolated strings.  Please convert to semantic logging.

File: src/Services/src/CfCli/CfCliService.cs (hunk 1 above)
Comment (atwoosnam) [reply src/Services/src/CfCli/CfCliService.cs:107]: Ok yeah this same thought crossed my mind in the moment -- first of all, I don't think it's crucial to include this `see cref` at all (I think the summary would be just fine without it), so I'm happy to get rid of it if it's unconventional.

These docs are mainly just for our own benefit at the moment, but if an external client did want to use this method, what exactly would the downside be with documenting an internal variable like this? Would it just seem pointless if that value is inaccessible?

Comment (ccheetham) [reply src/Services/src/CfCli/CfCliService.cs:107]: Correct.

**Review by ccheetham — APPROVED** [review]
lgtm

**Pull request merged**